packages/caravan-fees/src/rbf.ts

```diff
@@ -0,0 +1,360 @@
+ *   availableUTXOs,
+ *   1, // change output index
+ *   'bc1q...', // new change address (optional)
+ *   Network.MAINNET,
+ *   'P2WSH',
+ *   2, // required signers
+ *   3  // total signers
+ * );
+ *
+ * @see https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki BIP 125: Opt-in Full Replace-by-Fee Signaling
+ * @see https://developer.bitcoin.org/devguide/transactions.html#locktime-and-sequence-number Bitcoin Core's guide on locktime and sequence numbers
+ */
+export const createAcceleratedRbfTransaction = (
+  originalTx: string,
+  availableInputs: UTXO[],
+  network: Network,
+  dustThreshold: number,
+  scriptType: string,
+  requiredSigners: number,
+  totalSigners: number,
+  changeIndex?: number,
+  changeAddress?: string,
+): string => {
+  if (changeIndex === undefined && !changeAddress) {
+    throw new Error(
+      "Either changeIndex or changeAddress must be provided for handling the change output.",
+    );
+  }
+
+  // Step 1: Analyze the original transaction
+  const txAnalyzer = new TransactionAnalyzer({
+    txHex: originalTx,
+    network,
+    dustThreshold,
+    targetFeeRate: 0, // We'll use the analyzer's recommendation
+    absoluteFee: 0, // This will be calculated by the analyzer
+    availableUtxos: availableInputs,
+    requiredSigners,
+    totalSigners,
+    addressType: scriptType,
+    changeOutputIndex: changeIndex,
+  });
+
+  const analysis = txAnalyzer.analyze();
+
+  // Step 2: Verify RBF possibility and suitability
+  if (!analysis.canRBF) {
+    throw new Error(
+      "RBF is not possible for this transaction. Ensure at least one input has a sequence number < 0xfffffffe.",
+    );
+  }
+
+  if (analysis.recommendedStrategy !== FeeBumpStrategy.RBF) {
+    console.warn(
+      "RBF is not the recommended strategy for this transaction. Consider using the recommended strategy: " +
+        analysis.recommendedStrategy,
+    );
+  }
+
+  // Step 3: Create a new transaction template
+  const newTxTemplate = new BtcTransactionTemplate({
+    inputs: [],
+    outputs: [],
+    targetFeeRate: Number(txAnalyzer.rbfFeeRate), // Use RBF-specific fee rate from analyzer
+    dustThreshold: Number(txAnalyzer.getDustThreshold),
+    network,
+    scriptType,
+    requiredSigners,
+    totalSigners,
+  });
+
+  // Step 4: Add all non-change outputs from the original transaction
+  txAnalyzer.outputs.forEach((output, index) => {
+    if (index !== changeIndex) {
+      newTxTemplate.addOutput(
+        new BtcTxOutputTemplate({
+          address: output.address,
+          amountSats: output.value as Satoshis,
+          type: TxOutputType.DESTINATION,
+        }),
+      );
+    }
+  });
+
+  // Step 5: Add all inputs from the original transaction
+  txAnalyzer.getInputTemplates().forEach((inputTemplate) => {
+    const input = availableInputs.find(
+      (utxo) =>
+        utxo.txid === inputTemplate.txid && utxo.vout === inputTemplate.vout,
+    );
+    if (!input) {
+      throw new Error(
```

Comment:
Technically you only need to spend one of the original inputs, so if you don't have all of the inputs in your available inputs, I don't think you need to throw.

Reply:
[corrected](https://github.com/caravan-bitcoin/caravan/pull/114/commits/a215dea59e6ac918e960757d5dd6f74ed09ae55d#diff-47122bb461c82a41c84d8866adf0cb005bf7f573a2afb8558f670a419bed146aR327)